### PR TITLE
Task/sapnamysore/tlt 1870/cleanup migrations

### PR DESCRIPTION
**Pull request by sapnamysore**
This PR includes cleanup of existing migrations for manage people that had references to some  altered columns.  
Has new migrations to create the tables for manage people and also populating the values((based on current prod data). It also populates the lti_permissions tables with values for manage people app. 

Will update the wiki with the roles information in the DB.

### CODE REVIEW

**Comment by elliottyates** [comment]
@elliottyates

File: manage_people/migrations/0001_mp_initial.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+def create_lti_permissions(apps, schema_editor):
+    LtiPermission = apps.get_model('lti_permissions', 'LtiPermission')
+    fields = ('permission', 'school_id', 'canvas_role', 'allow')
+
+    for permission in LTI_PERMISSIONS_DATA:
+        LtiPermission.objects.create(**dict(zip(fields, permission)))
+
+def reverse_permissions_load(apps, schema_editor):
+    LtiPermission = apps.get_model('lti_permissions', 'LtiPermission')
+    LtiPermission.filter(permission='manage_people').delete()
```
Comment (elliottyates) [inline manage_people/migrations/0001_mp_initial.py]: `LtiPermission.objects.filter()`

File: manage_people/migrations/0001_mp_initial.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    (9, 'TeacherEnrollment', False),
+    (10, 'Guest', True),
+    (11, 'Course Support Staff', False),
+    (12, 'Teaching Staff', False),
+    (14, 'Shopper', False),
+    (15, 'ObserverEnrollment', False)
+]
+LTI_PERMISSIONS_DATA = [
+    ('manage_people', '*', 'Account Observer', True),
+	('manage_people', '*', 'AccountAdmin', True),
```
Comment (elliottyates) [inline manage_people/migrations/0001_mp_initial.py:24]: Do we have variations on `AccountAdmin` that need to be used (e.g. for QA)? or perhaps we can fix QA to match the string in prod, if we haven't already?

Comment (elliottyates) [inline manage_people/migrations/0001_mp_initial.py:24]: (i.e. when I ran it, I had to add a variation "Account admin" for my account in canvas.dev)

File: manage_people/migrations/0002_manage_people_uses_coursemanager_user_roles.py [hunk 3]
```diff
@@ -1,137 +0,0 @@
...
-    """
-    For each role in school_allowed_role, we need to:
-    * get the canvas_roles entry it refers to
-    * read the canvas_roles.canvas_role value
-    * find the matching user_role_id for that canvas_role from our canned
-      manage_people_role data we're populating during this migration
-    * replace the existing canvas_role_id with the new user_role_id
-
-    NOTE: this expects to be run before the schema change to rename
-          school_allowed_role.canvas_role_id to
```
Comment (elliottyates) [inline manage_people/migrations/0002_manage_people_uses_coursemanager_user_roles.py:71]: Looks like `models.SchoolAllowedRole.__unicode__()` throws an error because there is no canvas_role_id field. I think we actually missed this in our initial work updating icommons_lti_tools.course_add_people, but perhaps we can fix this method now? If not, we can do it as a separate PR.

File: manage_people/migrations/0002_manage_people_uses_coursemanager_user_roles.py [hunk 4]
```diff
@@ -1,137 +0,0 @@
...
-        migrations.RunPython(
-            code=update_school_allowed_role,
-            reverse_code=revert_school_allowed_role,
-        ),
-        migrations.RenameField(
-            model_name='schoolallowedrole',
-            old_name='canvas_role_id',
-            new_name='user_role_id',
-        ),
-        migrations.AlterUniqueTogether(
```
Comment (elliottyates) [inline manage_people/migrations/0002_manage_people_uses_coursemanager_user_roles.py:126]: In the current state Django won't respect the unique together that the SchoolAllowedRole model specifies.

Running `python manage.py migrate` gives:

```
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
```

And:

```
  (canvas_course_admin_tools)vagrant@vagrant-ubuntu-trusty-64:~/tlt/canvas_course_admin_tools(task/sapnamysore/tlt-1870/cleanup-migrations) $ python manage.py makemigrations
Migrations for 'manage_people':
  0003_auto_20160518_1418.py:
    - Alter field user_role on schoolallowedrole
    - Alter unique_together for schoolallowedrole (1 constraint(s))
```

For reference, here's what Django creates.

```

class Migration(migrations.Migration):

    dependencies = [
        ('manage_people', '0002_mp_school_allowed_roles'),
    ]

    operations = [
        migrations.AlterField(
            model_name='schoolallowedrole',
            name='user_role',
            field=models.ForeignKey(to='manage_people.ManagePeopleRole'),
        ),
        migrations.AlterUniqueTogether(
            name='schoolallowedrole',
            unique_together=set([('school_id', 'user_role')]),
        ),
    ]

```

After running that migration, Django observes the unique together constraint.

**Comment by elliottyates** [comment]
Very ncie and clean compared to the old code! Awesome! Looks like it needs a few tweaks to work with the existing models as expected and to enable the reverse migrations. After we get those cleaned up, it looks good!

File: manage_people/migrations/0002_manage_people_uses_coursemanager_user_roles.py (hunk 3 above)
Comment (sapnamysore) [inline manage_people/migrations/0002_manage_people_uses_coursemanager_user_roles.py:71]: Yes, lets handle that separately so I can push this thru. WIll add it to trello

File: manage_people/migrations/0001_mp_initial.py (hunk 2 above)
Comment (sapnamysore) [inline manage_people/migrations/0001_mp_initial.py:24]: Not adding the non-prod role into the migration for now. Maybe we should - as we already do for some tools.
